keel-core-test/src/main/kotlin/com/netflix/spinnaker/keel/persistence/DeliveryConfigRepositoryTests.kt

```diff
@@ -532,6 +534,26 @@ abstract class DeliveryConfigRepositoryTests<T : DeliveryConfigRepository, R : R
               expectThat(pausedRepository.getPause(PauseScope.RESOURCE, resource.id)).isNull()
             }
           }
+
+          test("deletes any older versions of resources") {
+            with(deliveryConfig.resources.first()) {
+              val newResourceVersion = resource(
+                kind = kind,
+                id = spec.id,
+                application = application
+              )
+              resourceRepository.store(newResourceVersion)
+            }
```

Comment:
Modify and save one of the resources so a version 2 will now be attached to the delivery config